azurerm/resource_arm_sql_server.go

```diff
@@ -60,101 +59,111 @@ func resourceArmSqlServer() *schema.Resource {
 	}
 }
 
-func resourceArmSqlServerCreate(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*ArmClient)
-	rivieraClient := client.rivieraClient
+func resourceArmSqlServerCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	sqlServersClient := meta.(*ArmClient).sqlServersClient
 
-	tags := d.Get("tags").(map[string]interface{})
-	expandedTags := expandTags(tags)
-
-	createRequest := rivieraClient.NewRequest()
-	createRequest.Command = &sql.CreateOrUpdateServer{
-		Name:                       d.Get("name").(string),
-		Location:                   d.Get("location").(string),
-		ResourceGroupName:          d.Get("resource_group_name").(string),
-		AdministratorLogin:         azure.String(d.Get("administrator_login").(string)),
-		AdministratorLoginPassword: azure.String(d.Get("administrator_login_password").(string)),
-		Version:                    azure.String(d.Get("version").(string)),
-		Tags:                       *expandedTags,
-	}
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	location := d.Get("location").(string)
+	admin := d.Get("administrator_login").(string)
+	adminPw := d.Get("administrator_login_password").(string)
+	fqdn := d.Get("fully_qualified_domain_name").(string)
 
-	createResponse, err := createRequest.Execute()
-	if err != nil {
-		return fmt.Errorf("Error creating SQL Server: %s", err)
+	version := d.Get("version").(string)
+	var serverVersion sql.ServerVersion
+	if version == string(sql.OneTwoFullStopZero) {
+		serverVersion = sql.OneTwoFullStopZero
+	}
+	if version == string(sql.TwoFullStopZero) {
+		serverVersion = sql.TwoFullStopZero
 	}
-	if !createResponse.IsSuccessful() {
-		return fmt.Errorf("Error creating SQL Server: %s", createResponse.Error)
+	if serverVersion == "" {
+		return fmt.Errorf("Invalid server version provided. It must be one of 12.0 or 2.0, passed as string: %s", version)
+	}
```

Comment:
[we can remove this in favour of adding validation here](https://github.com/terraform-providers/terraform-provider-azurerm/pull/189/files#diff-73132f55101d4243316253ba03f30742R36)

i.e.

```
ValidateFunc: validation.StringInSlice([]string {
  string(sql.OneTwoFullStopZero),
  string(sql.TwoFullStopZero),
}, true)
```